File: stopwatch/java-mutable/StopWatch.java
public class StopWatch {

    private long start;
    private long end;
    
    public void start() {
        start = System.currentTimeMillis();
    }
    
    public void stop() {
        end = System.currentTimeMillis();
    }
    
    public long elapsed() {
        return end - start;
    }
}

File: stopwatch/kt-types/stopwatch.kt
sealed class Stopwatch {

    class Started() : Stopwatch() {
        internal val startTime = System.currentTimeMillis()
    }
    data class Stopped(
        val startedStopwatch: Started,
    ) : Stopwatch() {
        private val stopTime = System.currentTimeMillis()
        val elapsed: Long = stopTime - startedStopwatch.startTime
    }
}

fun main() {
    val startedStopwatch = Stopwatch.Started()
    Thread.sleep(1000)
    val stoppedStopwatch = Stopwatch.Stopped(startedStopwatch)
    println(stoppedStopwatch.elapsed)
}
